Plugin.pm

```diff
@@ -1491,6 +1484,65 @@ sub QobuzGetTracks {
 	}, $albumId);
 }
 
+sub _albumPerformers {
+	my ($client, $performers, $trackCount, $items) = @_;
+
+	my @uniquePerformers;
+	my %seen = ();
+	my $tracks;
+
+	foreach my $disc (sort(keys %$performers)) {
+		my %discAdded = ();
+		foreach my $item (@{$performers->{$disc}}) {
+			push @{$tracks->{$item->{'name'}}->{'tracks'}}, " " . cstring($client, 'DISC') . " $disc" . cstring($client, 'COLON') . " " unless $discAdded{$item->{'name'}}++ || scalar keys %$performers == 1;
+			push @{$tracks->{$item->{'name'}}->{'tracks'}}, $item->{'track'};
+			delete $item->{'track'};
+			push(@uniquePerformers, $item) unless $seen{$item->{'name'}}++;
+		}
+	}
+
+	if ( scalar @uniquePerformers ) {
+		foreach my $item (@uniquePerformers) {
+			my @tracks = @{$tracks->{$item->{'name'}}->{'tracks'}};
+			my $creditCount = scalar @tracks - (scalar keys %$performers == 1 ? 0 : scalar keys %$performers);
+			if ( @tracks && scalar $creditCount < $trackCount ) {
+				$item->{'name'} .= " ( ";
+
+				# collapse the track list so that, eg, 1,2,3,5,7,8,9,11,12 becomes 1-3, 5, 7-9, 11-12 and add punctuation to make multi-disc albums somewhat intelligible
+				# there's probably a much more perly way of doing this...
+				my $sep = "-";
+				my $o;
+				for my $i ( 0 .. $#tracks ) {
+					if ( looks_like_number($tracks[$i]) && $tracks[$i+1] == $tracks[$i]+1) {
+						$o .= "$tracks[$i]$sep" if $sep;
+						$sep = undef;
+					} else {
+						$sep = "-";
+						if ( looks_like_number($tracks[$i]) && looks_like_number($tracks[$i+1]) ) {
+							$o .= "$tracks[$i], ";
+						} elsif ( looks_like_number($tracks[$i]) && $tracks[$i+1] && !looks_like_number($tracks[$i+1]) ) {
+							$o .= "$tracks[$i]; ";
+						} else {
+							$o .= "$tracks[$i]";
+						}
+					}
```

Comment:
For readability and to save at least two CPU cycles I'd store `$tracks[$I]` and `$tracks[$I+1]` in variables, as you're using these two values all over the place. Something like `$currentValue` and `$nextValue`. You could even do the same with all the `looks_like_number` evaluations. Should give a more readable block of code. Because right now I'm having a hard time figuring out all the dollar signs and increments and what not.

Reply:
One more thing: you could remove all the checks for `looks_like_number($tracks[$I])` and replace them with a very first check against this and return early if it failed.

Reply:
Could you please tweak this? Then we'd be good to go?